monitoring/asset_proof_collector.go

```diff
@@ -0,0 +1,110 @@
+package monitoring
+
+import (
+	"context"
+	"errors"
+	"sync"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	assetProofSizesHistogram = "asset_proofs_sizes_histogram"
+)
+
+// assetProofCollector is a Prometheus collector that exports proof related data
+// of all taproot assets.
+type assetProofCollector struct {
+	collectMx sync.Mutex
+
+	cfg      *PrometheusConfig
+	registry *prometheus.Registry
+
+	proofsHistogram prometheus.Histogram
+}
+
+func newAssetProofCollector(cfg *PrometheusConfig,
+	registry *prometheus.Registry) (*assetProofCollector, error) {
+
+	if cfg == nil {
+		return nil, errors.New("asset proof collector prometheus cfg " +
+			"is nil")
+	}
+
+	if cfg.AssetStore == nil {
+		return nil, errors.New("asset proof collector asset store is " +
+			"nil")
+	}
+
+	return &assetProofCollector{
+		cfg:      cfg,
+		registry: registry,
+		proofsHistogram: prometheus.NewHistogram(
```

Comment:
Shouldn't we use `newProofSizesHistogram` here? Seems like the values are different here as well for the buckets...

Reply:
oh yes, will use that here too